test/metabase/models/pulse_channel_test.clj

```diff
@@ -415,3 +416,26 @@
                  :last_name   "Toucan"
                  :common_name "Rasta Toucan"}]))
              (:recipients (hydrate channel :recipients)))))))
+
+(deftest validate-email-domains-check-user-ids-match-emails
+  (testing `pc/validate-email-domains
+    (testing "should check that User `:id` and `:email`s match for User `:recipients`"
+      (let [input {:recipients [{:email "rasta@metabase.com"
+                                 :id    (mt/user->id :rasta)}]}]
+        (is (= input
+               (pc/validate-email-domains input))))
+      (testing "Throw Exception if User does not exist"
+        ;; should validate even if `:email` isn't specified
+        (doseq [input [{:id Integer/MAX_VALUE}
+                       {:email "rasta@example.com"
+                        :id    Integer/MAX_VALUE}]]
```

Comment:
Nit: seems like the map values aren't horizontally aligned, but there are also multiple spaces after `:id` here.  I would go with one or the other.

Reply:
they're definitely horizontally aligned. Maybe it just looks off because there are quote marks on the value for `:email`

Reply:
Blah, my mistake.  `{:id Integer/MAX_VALUE}` is a completely different map.